docs/reference/contributing/target/us_ticker.md

```diff
@@ -0,0 +1,69 @@
+### Microsecond ticker
+
+Implementing the microsecond ticker enables Mbed OS to perform operations that require precise timing. You can use this API to shedule events, record elapsed time and perform submillisecond delays.
+
+<span class="warnings">**Warning:** We are changing the microsecond ticker HAL API in an upcoming release of Mbed OS. You can find details on how it may affect you in the [implementing the microsecond ticker API](#implementing-the-microsecond-ticker-api) section.
+
+#### Assumptions
+
+##### Defined behavior
+
+- Has a reported frequency between 250KHz and 8MHz.
+- Has a counter that is at least 16 bits wide.
+- The function `ticker_init` is safe to call repeatedly.
+- The function `ticker_init` allows the ticker to keep counting and disables the ticker interrupt.
+- Ticker frequency is nonzero, and the counter is at least 8 bits.
+- The ticker rolls over at (1 << bits) and continues counting starting from 0.
+- The ticker counts at the specified frequency plus or minus 10%.
+- The ticker increments by 1 each tick.
+- The ticker interrupt fires only when the ticker times increments to or past the value set by `ticker_set_interrupt`.
+- It is safe to call `ticker_set_interrupt` repeatedly before the handler is called.
+- The function `ticker_fire_interrupt` causes `ticker_irq_handler` to be called immediately from interrupt context.
+- The ticker operations `ticker_read`, `ticker_clear_interrupt`, `ticker_set_interrupt` and `ticker_fire_interrupt` take less than 20us to complete.
+
+##### Undefined behavior
+
+- Calling any function other than `ticker_init` before the initialization of the ticker.
+- Whether `ticker_irq_handler` is called a second time if the time wraps and matches the value set by `ticker_set_interrupt` again.
+- Calling `ticker_set_interrupt` with a value that has more than the supported number of bits.
+- Calling any function other than `us_ticker_init` after calling `us_ticker_free`.
+
+##### Notes
+
+Watch out for these common pitfalls when implementing this API:
```

Comment:
Could we change "pitfalls" to "trouble areas" in both files and still keep the meaning you want? I'd like to replace "pitfalls" with something less idiomatic.